src/core/api/chart_repository.go

```diff
@@ -365,62 +441,6 @@ func (cra *ChartRepositoryAPI) requireNamespace(namespace string) bool {
 	return true
 }
 
-// Check if the related access match the expected requirement
```

Comment:
Why not update the `requireAccess` to accept resource and `Action`?
I don't think @steven-zou is gonna like this your refactor here.

We can merge it but please open a new issue.

Reply:
refactor `requireAccess` using security.Context new method